WikiWatershed/model-my-watershed
# Upgrade jQuery

## Overview

Upgrades jQuery to 3.3.1 for security reasons. Also adjusts various deprecated calls in accordance with [the migration guide](https://jquery.com/upgrade-guide/3.0/).

Connects #2613 

### Notes

I expect subtle edge cases to be discovered as side effects of this in the near future.

## Testing Instructions

  * Check out this branch
  * The last commit removes `jquery-migrate`, which logs deprecated features to the console. Do `git checkout HEAD~1` to get a version of this branch that still has it.
  * Run `./scripts/npm.sh install` to get the new jQuery
  * Go to [:8000/](http://localhost:8000/) and verify _the entire app_ with the console open and ensure everything works and there are no deprecation warnings
  * Go to [:8000/?bigcz](http://localhost:8000/?bigcz) and verify _the entire app_ with the console open and ensure everything works and there are no deprecation warnings
